app/org/sagebionetworks/bridge/play/controllers/UserManagementController.java

```diff
@@ -41,6 +40,27 @@ public Result createUser() throws Exception {
         return createdResult(UserSessionInfo.toJSON(userSession));
     }
 
+    /**
+     * admin api used to create consent/not-consent user for given study
+     * nearly identical to createUser() one
+     * @param studyId
+     * @return
+     * @throws Exception
+     */
+    public Result createUserWithStudyId(String studyId) throws Exception {
+        getAuthenticatedSession(ADMIN);
+        Study study = studyService.getStudy(studyId);
+
+        JsonNode node = requestToJSON(request());
+        StudyParticipant participant = parseJson(request(), StudyParticipant.class);
+
+        boolean consent = JsonUtils.asBoolean(node, CONSENT_FIELD);
+
+        UserSession userSession = userAdminService.createUser(study, participant, null, false, consent);
```

Comment:
Because you are creating a different user from the caller, you probably shouldn't return the session, unless you anticipate using this in tests.